src/volatile_memory.rs

```diff
@@ -538,6 +538,32 @@ impl<'a, B: BitmapSlice> VolatileSlice<'a, B> {
         }
     }
 
+    /// Returns a subslice of this [`VolatileSlice`] of length at most `count`.
+    ///
+    /// If `count` is larger than the length of `self`, returns a copy of `self`
+    ///
+    /// ```rust
+    /// # use vm_memory::VolatileSlice;
+    /// let mut arr = [1, 2, 3, 4, 5];
+    /// let slice = VolatileSlice::from(arr.as_mut_slice());
+    ///
+    /// assert_eq!(slice.truncate(3).len(), 3);
+    /// assert_eq!(slice.truncate(10).len(), 5);
+    /// ```
+    pub fn truncate(&self, count: usize) -> VolatileSlice<'a, B> {
```

Comment:
Truncate somehow implies (to me) that we're updating self because of how `truncate` on Vec works: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.truncate.

Shouldn't we be able to obtain the same result by calling `subslice` with offset = 0 & count as the size we want to "truncate" to?

Reply:
Argh, true. Naming is hard. Maybe `.truncated()` instead (e.g. with an extra 'd')?

> Shouldn't we be able to obtain the same result by calling subslice with offset = 0 & count as the size we want to "truncate" to?

It'd need to be count = min(size we want to be, length). I didn't like it because subslice does a million bounds checks that are all unnecessary for these arguments. But I should probably actually benchmark it/see if the compiler manages to optimize it away

Reply:
Damn, criterion indeed says there's no difference between them. Guess rustc is better than I thought at optimizing xP

So my only argument is "it avoids .unwrap()-ing something that we know can never fail"